pymatgen/transformations/standard_transformations.py

```diff
@@ -280,10 +282,19 @@ def from_boundary_distance(
                 min_boundary_dist
                 / np.array([struct_scaled.lattice.d_hkl(plane) for plane in [[1, 0, 0], [0, 1, 0], [0, 0, 1]]])
             )
-            if np.count_nonzero(min_expand_scaled) == 0:
+            if np.count_nonzero(min_expand_scaled) == 0 and len(struct_scaled) <= max_atoms:
                 return SupercellTransformation(scaling_matrix)
 
-        return SupercellTransformation(np.eye(3) + np.diag(min_expand))
+        scaling_matrix = np.eye(3) + np.diag(min_expand)
+        struct_scaled = structure.make_supercell(scaling_matrix, in_place=False)
+        if len(struct_scaled) <= max_atoms:
+            return SupercellTransformation(scaling_matrix)
+
+        raise AttributeError(
```

Comment:
It shouldn't be an AttributeError. This has a special meaning in python. In this case, it should be a RuntimeError.